### PR TITLE
Added the ability to choose puck configuration from the general settings

### PR DESCRIPTION
**Pull request by druiz17**
Added the ability to choose puck configuration from the general settings

### CODE REVIEW

**Comment by druiz17** [comment]
Test plan
- to set the configuration type, go to the very bottom of the general setting.
- the configuration types are Auto, Feet, FeetAndHips, FeetHipsAndChest

2 Pucks enabled
- Auto - should configure the feet
- Feet - should configure the feet
- FeetAndHips - should not configure anything
-FeetHipsAndChest - should not configure anything

3 Pucks enabled
- Auto - should configure the feet and hips
- Feet - should configure the feet
- FeetAndHips - should configure the feet and hips
-FeetHipsAndChest - should not configure anything

4 Pucks enabled
- Auto - should configure the feet, hips, and chest
- Feet - should configure the feet
- FeetAndHips - should configure the feet and hips
-FeetHipsAndChest - should configure the feet, hips, and chest

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0f0bf79367081ee56f0c49128b742a1c1b9e34f3 available at: https://gist.github.com/e7242a27402b34f8fad4b74a1d91aa0c<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10436-0f0bf79.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10436-0f0bf79.dmg

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 1]
```diff
@@ -544,6 +557,74 @@ void ViveControllerManager::InputDevice::hapticsHelper(float deltaTime, bool lef
...
+            break;
+    }
+    return currentConfig;
+}
+
+void ViveControllerManager::InputDevice::setConfigFromString(const QString& value) {
+    if (value ==  "Auto") {
+        _preferedConfig = Config::Auto;
+    } else if (value == "Feet") {
+            _preferedConfig = Config::Feet;
```
Comment (sethalves) [inline plugins/openvr/src/ViveControllerManager.cpp]: bad indent

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 2]
```diff
@@ -544,6 +557,74 @@ void ViveControllerManager::InputDevice::hapticsHelper(float deltaTime, bool lef
...
+    Settings settings;
+    settings.beginGroup("PUCK_CONFIG");
+    {
+        settings.setValue(QString("configuration"), (int)_preferedConfig);
+    }
+    settings.endGroup();
+}
+
+QString ViveControllerManager::InputDevice::configToString() {
+    QString currentConfig = "";
```
Comment (sethalves) [inline plugins/openvr/src/ViveControllerManager.cpp]: empty string is the default for a QString

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 3]
```diff
@@ -544,6 +557,74 @@ void ViveControllerManager::InputDevice::hapticsHelper(float deltaTime, bool lef
...
+        _preferedConfig = Config::FeetHipsAndChest;
+    }
+}
+
+void ViveControllerManager::InputDevice::createPreferences() {
+    loadSettings();
+    auto preferences = DependencyManager::get<Preferences>();
+    static const QString VIVE_PUCKS_CONFIG { "Vive Pucks Configuration" };
+
+    {
```
Comment (sethalves) [inline plugins/openvr/src/ViveControllerManager.cpp]: putting this code in a lexical block seems unnecessary

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7687617e912719c4de3455992492dcb73e688c16 available at: https://gist.github.com/ad596316865567b19bef3ba3fccfe80e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10436-7687617.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10436-7687617.dmg

**Comment by anshuman64** [comment]
QA Fail. Looks like you have the following switched:

Chest <-> Right foot
Hips <-> Left foot

**Comment by anshuman64** [comment]
Well deserved QA Pass! :D

**Pull request merged**